### PR TITLE
Add order details new column picker

### PR DESCRIPTION
**Pull request by poulch**
closes https://github.com/saleor/saleor-dashboard/issues/3754
### Pull Request Checklist

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `data-test-id` are added for new elements
4. [ ] The changes are tested in Chrome/Firefox/Safari browsers and in light/dark mode
5. [ ] Your code works with the latest stable version of the core
6. [ ] I added changesets and [read good practices](/.changeset/README.md)

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
APPS_MARKETPLACE_API_URI=https://apps.staging.saleor.io/api/v2/saleor-apps

### Do you want to run more stable tests?

To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [x] stable
2. [ ] app
3. [ ] attribute
4. [ ] category
5. [ ] collection
6. [ ] customer
7. [ ] giftCard
8. [ ] homePage
9. [ ] login
10. [ ] menuNavigation
11. [ ] navigation
12. [x] orders
13. [ ] pages
14. [ ] payments
15. [ ] permissions
16. [ ] plugins
17. [ ] productType
18. [ ] products
19. [ ] sales
20. [ ] shipping
21. [ ] translations
22. [ ] variants
23. [ ] vouchers

CONTAINERS=5

### CODE REVIEW

File: src/config.ts [hunk 1]
```diff
@@ -61,6 +61,8 @@ export interface AppListViewSettings {
   [ListViews.WEBHOOK_LIST]: ListSettings;
   [ListViews.TRANSLATION_ATTRIBUTE_VALUE_LIST]: ListSettings;
   [ListViews.GIFT_CARD_LIST]: ListSettings;
+  [ListViews.ORDER_DETAILS_LIST]: ListSettings;
+  [ListViews.ORDER_DRAFT_DETAILS_LIST]: ListSettings;
```
Comment (Droniu) [inline src/config.ts:65]: **suggestion**: I would consider naming them `ORDER_DETAILS` and `ORDER_DRAFT_DETAILS` - they are not strictly list views, they just have a datagrid section.

File: src/orders/components/OrderDraftDetailsDatagrid/datagrid.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {
   thumbnailCell,
 } from "@dashboard/components/Datagrid/customCells/cells";
 import { GetCellContentOpts } from "@dashboard/components/Datagrid/Datagrid";
-import { useEmptyColumn } from "@dashboard/components/Datagrid/hooks/useEmptyColumn";
+import {} from "@dashboard/components/Datagrid/hooks/useEmptyColumn";
```
Comment (Droniu) [inline src/orders/components/OrderDraftDetailsDatagrid/datagrid.ts:11]:
```suggestion
```

File: src/orders/components/OrderDraftDetailsDatagrid/datagrid.ts [hunk 3]
```diff
@@ -21,62 +21,52 @@ import { useOrderLineDiscountContext } from "@dashboard/products/components/Orde
...
+    width: 150,
+  },
+  {
+    id: "total",
+    title: intl.formatMessage(columnsMessages.total),
+    width: 150,
+  },
+  {
+    id: "status",
+    title: "Status",
```
Comment (Droniu) [inline src/orders/components/OrderDraftDetailsDatagrid/datagrid.ts:66]: **issue**: untranslated message. I see it was like that before, but I think this should be fixed

**Review by Droniu — APPROVED** [review]

**Review by Droniu — APPROVED** [review]

**Review by krzysztofzuraw — DISMISSED** [review]

**Comment by wojteknowacki** [comment]
Infinite loader on variants when finalizing draft order
https://github.com/saleor/saleor-dashboard/assets/124166231/488df176-5b4a-4ab7-bd26-f0b0b6c2141d

**Review by krzysztofzuraw — DISMISSED** [review]

**Comment by wojteknowacki** [comment]
Uploading Nagranie z ekranu 2023-07-24 o 09.02.04.mov…

**Comment by wojteknowacki** [comment]
https://github.com/saleor/saleor-dashboard/assets/124166231/1c568aff-871f-4c72-9d6a-75444aca09ce

**Review by krzysztofzuraw — APPROVED** [review]

**Review by wojteknowacki — APPROVED** [review]

**Pull request merged**